modules/KalturaSupport/components/share/share.js

```diff
@@ -616,7 +616,7 @@
 					res = document.URL;
 				}
 			}
-			return res;
+			return kWidget.sanitize(decodeURIComponent(res));
```

Comment:
do we need to encode the url again?

Reply:
no, the decodeURIComponent replaces each escape sequence in the encoded with the character that it represents in ascii. if it's not an escaping char, it moves on. for example:
https://www.kaltura.com/.../embed/dynamic?%22%3E%3Cimg%20src=x%20onerror=prompt(4)%3E
would turn into:
https://www.kaltura.com/.../embed/dynamic?"><img src=x onerror=prompt(4)>
by doing so, the sanitizer would be able to replace any escaping chars with empty strings. without that decoding stage, sanitizer won't work in cases where there is encoded uri. if the uri is not encoded (most cases), the decoder does nothing and returns the uri string as-is.

Reply:
i think encode it again it safer. if it was encoded we have to keep it encoded. if not we just encode the url, what the side effect? @OrenMe wdyt?